app/data_models/questionnaire_store.py

```diff
@@ -120,48 +204,55 @@ def _remove_old_supplementary_lists_and_answers(
         :param new_data - the new supplementary data for comparison
         """
         deleted_list_item_ids: set[str] = set()
-        for list_name, mappings in self.supplementary_data_store.list_lookup.items():
+        for (
+            list_name,
+            mappings,
+        ) in self.data_stores.supplementary_data_store.list_lookup.items():
             if list_name in new_data.get("items", {}):
                 new_identifiers = [
                     item["identifier"] for item in new_data["items"][list_name]
                 ]
                 for identifier, list_item_id in mappings.items():
                     if identifier not in new_identifiers:
-                        self.list_store.delete_list_item(list_name, list_item_id)
+                        self.data_stores.list_store.delete_list_item(
+                            list_name, list_item_id
+                        )
                         deleted_list_item_ids.add(list_item_id)
             else:
-                self.list_store.delete_list(list_name)
+                self.data_stores.list_store.delete_list(list_name)
                 deleted_list_item_ids.update(mappings.values())
-        self.answer_store.remove_all_answers_for_list_item_ids(*deleted_list_item_ids)
+        self.data_stores.answer_store.remove_all_answers_for_list_item_ids(
+            *deleted_list_item_ids
+        )
 
     def _deserialize(self, data: str) -> None:
         json_data = json_loads(data)
-        self.progress_store = ProgressStore(json_data.get("PROGRESS"))
+        self.data_stores.progress_store = ProgressStore(json_data.get("PROGRESS"))
         self.set_metadata(json_data.get("METADATA", {}))
-        self.supplementary_data_store = SupplementaryDataStore.deserialize(
+        self.data_stores.supplementary_data_store = SupplementaryDataStore.deserialize(
             json_data.get("SUPPLEMENTARY_DATA", {})
         )
-        self.answer_store = AnswerStore(json_data.get("ANSWERS"))
-        self.list_store = ListStore.deserialize(json_data.get("LISTS"))
-        self.response_metadata = json_data.get("RESPONSE_METADATA", {})
+        self.data_stores.answer_store = AnswerStore(json_data.get("ANSWERS"))
+        self.data_stores.list_store = ListStore.deserialize(json_data.get("LISTS"))
+        self.data_stores.response_metadata = json_data.get("RESPONSE_METADATA", {})
 
     def serialize(self) -> str:
         data = {
             "METADATA": self._metadata,
-            "ANSWERS": list(self.answer_store),
-            "SUPPLEMENTARY_DATA": self.supplementary_data_store.serialize(),
-            "LISTS": self.list_store.serialize(),
-            "PROGRESS": self.progress_store.serialize(),
-            "RESPONSE_METADATA": self.response_metadata,
+            "ANSWERS": list(self.data_stores.answer_store),
+            "SUPPLEMENTARY_DATA": self.data_stores.supplementary_data_store.serialize(),
+            "LISTS": self.data_stores.list_store.serialize(),
```

Comment:
Since this is serialising each thing in the data stores, I think a serialise method in the DataStores class which gets called here would make sense

Reply:
The only thing is, each of these `serialize` methods does something slightly different, depending on the store. For exa ple, `SuplementaryDataStore` one creates this object:
```python
return {
            "data": self._raw_data,
            "list_mappings": self._list_mappings,
        }
```

Reply:
I'd just wondered if it would make sense to have a `serialize` method in the datastores, which is essentially just this one, and saves you needing to do `self.data_stores.....serialize()` etc and makes it the responsibility of datastores, and the questionnaire store would then be able to just call that

Reply:
I get what you mean, we could do all of it in `data_stores` but in both `serialize` and `_deserialize` we change `QuestionnaireStore`'s `self._metadata` so I am not sure if we want to separate it out? `serialize` also calls `QuestionnaireStore`'s `self.set_metadata` method.

Reply:
I had to do couple more refactors to move it to `data_stores`.